Use `no-skip` flags to restore components `minimal` skips

brtrick opened the pull request:
Allow `--no-skip-*` flags with `rails new` to restore individual components
in a `--minimal` build.

For example, `rails new cool_app --minimal --no-skip-action-cable` will
create a standard minimal app with Action Cable functionality also included.

Allowed flags are:

    --no-skip-action-cable
    --no-skip-action-mailer
    --no-skip-action-mailbox
    --no-skip-action-text
    --no-skip-active-job
    --no-skip-active-storage
    --no-skip-bootsnap
    --no-skip-dev-gems
    --no-skip-javascript
    --no-skip-jbuilder
    --no-skip-system-test
    --no-skip-hotwire

Commit also makes `:skip_dev_gems` a `class_option` for `AppBase`
and includes tests for each flag.
<!-- ### Summary -->

<!-- Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together. -->

<!-- ### Other Information -->

<!-- If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

If you are updating any of the CHANGELOG files or are asked to update the
CHANGELOG files by reviewers, please add the CHANGELOG entry at the top of the file.

Finally, if your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

Thanks for contributing to Rails! -->

<!--
Note: Please avoid making *Draft* pull requests, as they still send
notifications to everyone watching the Rails repo.
Create a pull request when it is ready for review and feedback
from the Rails team :).
-->

jonathanhefner reviewed (commented):
Thank you for tackling this!

One concern that needs to be addressed is how to handle option interactions.  For example, [`skip_javascript` implies `skip_hotwire`](https://github.com/rails/rails/blob/3fd0f848bda056bda3e09b78c77bc114a09bba55/railties/lib/rails/generators/app_base.rb#L333), so if `--minimal --no-skip-hotwire` is specified, what should we do?

One possibility would be to simply log when one option cancels another, e.g.

```ruby
return if options[:skip_hotwire]
if options[:skip_javascript]
  say "Skipping hotwire because javascript was skipped."
  return
end
```

brtrick commented on railties/lib/rails/generators/rails/app/app_generator.rb:
```diff
@@ -282,19 +282,21 @@ def initialize(*args)
-            skip_jbuilder: true,
-            skip_system_test: true,
-            skip_hotwire: true).freeze
+            skip_action_cable:   no_skip_args.include?("--no-skip-action-cable")   ? false : true,
+            skip_action_mailer:  no_skip_args.include?("--no-skip-action-mailer")  ? false : true,
+            skip_action_mailbox: no_skip_args.include?("--no-skip-action-mailbox") ? false : true,
+            skip_action_text:    no_skip_args.include?("--no-skip-action-text")    ? false : true,
+            skip_active_job:     no_skip_args.include?("--no-skip-active-job")     ? false : true,
+            skip_active_storage: no_skip_args.include?("--no-skip-active-storage") ? false : true,
+            skip_bootsnap:       no_skip_args.include?("--no-skip-bootsnap")       ? false : true,
+            skip_dev_gems:       no_skip_args.include?("--no-skip-dev-gems")       ? false : true,
+            skip_javascript:     no_skip_args.include?("--no-skip-javascript")     ? false : true,
+            skip_jbuilder:       no_skip_args.include?("--no-skip-jbuilder")       ? false : true,
+            skip_system_test:    no_skip_args.include?("--no-skip-system-test")    ? false : true,
+            skip_hotwire:        no_skip_args.include?("--no-skip-hotwire")        ? false : true
```
Love this solution! Thanks for the suggestion.

brtrick committed:
Use `no-skip` flags to restore components `minimal` skips

This commit enables the use of `--no-skip-*` flags with `rails new` to
restore components that would usually be skipped in a `--minimal` build.
For example,

    rails new cool_app --minimal --no-skip-action-cable

will create a standard minimal app with Action Cable functionality also
included. Allowed flags are:

    --no-skip-action-cable
    --no-skip-action-mailer
    --no-skip-action-mailbox
    --no-skip-action-text
    --no-skip-active-job
    --no-skip-active-storage
    --no-skip-bootsnap
    --no-skip-dev-gems
    --no-skip-javascript
    --no-skip-jbuilder
    --no-skip-system-test
    --no-skip-hotwire

Commit also makes `:skip_dev_gems` a `class_option` for `AppBase` and
includes tests for each flag.

brtrick commented:
@jonathanhefner This is ready for re-review. I set the skip options to insert at the beginning of option_argv so `--minimal` could be included anywhere in the command line; if you prefer to enforce cl order, let me know and I will insert them right before `--minimal` instead.

I'll take a stab at logging the option interactions in a separate pr.
